espnet/nets/pytorch_backend/ctc.py

```diff
@@ -25,13 +25,14 @@ def __init__(self, odim, eprojs, dropout_rate, ctc_type="warpctc", reduce=True):
         self.loss = None
         self.ctc_lo = torch.nn.Linear(eprojs, odim)
         self.probs = None  # for visualization
+        self.ctc_type = ctc_type
 
         # In case of Pytorch >= 1.7.0, CTC will be always builtin
-        self.ctc_type = (
-            ctc_type
-            if LooseVersion(torch.__version__) < LooseVersion("1.7.0")
-            else "builtin"
-        )
+        # self.ctc_type = (
+        #     ctc_type
+        #     if LooseVersion(torch.__version__) < LooseVersion("1.7.0")
+        #     else "builtin"
+        # )
```

Comment:
I think this part should be refined.
This part was made since warpctc does not support pytorch 1.7.0
gtn and k2 should work with Pytorch >= 1.7.0 while they would not work with some older Pytorch versions.
It would be great if you refine this condition and adding gtn and k2 unit test so that it could pass the CI check (we're checking PyTorch 1.0.1, 1.1.0, 1.2.0, 1.3.1, 1.4.0, 1.5.1, 1.6.0, and 1.7.1)

Reply:
I will try to refine this logic.